### DISCUSSION TITLE
Cannot use db push: ERROR: type "CIVILITY" already exists

### DISCUSSION THREAD

**Discussion by Extarys** [prisma/prisma#22558] [Q&A]
### Question

Every time I make a change to my schema, I have to manually delete the public schema inside postgres, delete my migration and redo it.

Else, when I run `yarn prisma db push --accept-data-loss`, I get:
```
Environment variables loaded from .env
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "test", schema "public" at "localhost:5432"
Error: ERROR: type "CIVILITY" already exists
   0: sql_schema_connector::apply_migration::migration_step
           with step=CreateEnum(EnumId(0))
             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:21
   1: sql_schema_connector::apply_migration::apply_migration
             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:10
   2: schema_core::state::SchemaPush
             at schema-engine/core/src/state.rs:436
```
I'm wondering what am I doing wrong.
Your insights are welcome. :)

(And happy new year!)

### How to reproduce (optional)

1. Create a schema with an enum and a random table
2. Create initial migration
3. Change something in the document (i.e. relationship, table)
4. Use `db push`

### Expected behavior (optional)

I can push any changes for quick prototyping.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Fedora
- Database: PostgreSQL/Yugabyte
- Node.js version: 20.10.0

- Run `prisma -v` to see your Prisma version and paste it
- 5.7.1

**Comment by ludralph** [comment]
Hi @Extarys 👋 

Thank you for raising this question.

I am unable to reproduce the issue using the instructions you provided. Can you please share a reproducible example in a repository?

Additionally, you would need to check your Prisma migration files to ensure that you are not trying to create an enum that already exists in the database. If the enum should exist, you may need to adjust your migration files to reflect the current state of the database. For example, in your `migration.sql` file, kindly check that the enum `Civility` is not duplicated.

**Comment by rogeriochaves** [comment]
hey there, I'm facing the same issue, Postgres on AWS RDS, not using migrations just `npx prisma db push`, with mysql it works fine, but with postgres seems like it always try to create the same enum types again and it fails

**Comment by rogeriochaves** [comment]
Okay, found the solution here: https://github.com/prisma/prisma/discussions/12452#discussioncomment-2478962

Add `&schema=<your schema>` to the pg connection string

**Comment by shiny** [comment]
facing same problem.

If I define an Enum in model, I can't run `db push` multiple times.

```
model User {
  id        String     @id
  email     String     @unique
  name      String?
  status    UserStatus @default(WaitingConfirmation)
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
}

enum UserStatus {
  Confirmed
  WaitingConfirmation
  Suspended
  Deleted
}
```

Error message blow

```
Error: ERROR: type "UserStatus" already exists
   0: sql_schema_connector::apply_migration::migration_step
           with step=CreateEnum(EnumId(0))
             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:21
   1: sql_schema_connector::apply_migration::apply_migration
             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:10
   2: schema_core::state::SchemaPush
             at schema-engine/core/src/state.rs:504
```

**Reply by ludralph** [reply]
Hi @Extarys 

Can you please open a new discussion if this is still an issue?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
